src/main/java/hudson/plugins/analysis/core/ParserResult.java

```diff
@@ -457,6 +458,15 @@ public String getLogMessages() {
         return StringUtils.defaultString(logMessage);
     }
 
+    /**
+     * repopulate the transient fileNameCache
+     */
+    private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
+        in.defaultReadObject();
+
+        this.fileNameCache = HashMultimap.create();
```

Comment:
This prevents a NPE, but will dramatically reduce the performance. 

Since the feature of mapping relative filenames to absolute filenames needs to be revised anyway I think is ok to apply this patch for now. I'm not sure how many people are aware of that feature anyway, since it is disabled by default.

Reply:
More importantly, it will fix JENKINS-33210 which is all I care about. :-) 

I can add a `populateFileNameCache()` after this which would rebuild the cache after deserialization.

Reply:
I am not sure how necessary that would be. Lookups all will hit `expandRelativePaths` which in turn uses `findFilesByScanningAllWorkspaceFiles`. And that checks whether the cache is empty and repopulates it if necessary.